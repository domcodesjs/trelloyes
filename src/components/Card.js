import React from 'react';

const Card = ({ title, content }) => {
  return (
    <div className='card'>
      <div>
        <h3>{title}</h3>
        <button type='button'>delete</button>
      </div>
      <p>{content}</p>
    </div>
  );
};

export default Card;
