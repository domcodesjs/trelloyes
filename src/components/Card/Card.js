import React from 'react';
import PropTypes from 'prop-types';

const Card = ({ title, content }) => {
  return (
    <div className='card'>
      <div>
        <h3>{title}</h3>
        <button type='button'>delete</button>
      </div>
      <p>{content}</p>
    </div>
  );
};

Card.propTypes = {
  title: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired,
};

export default Card;
