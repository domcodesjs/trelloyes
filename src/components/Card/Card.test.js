import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Card from './Card';

describe('Card Component', () => {
  // Smoke Test
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Card
        title='Title'
        content={'lorem ipsum something you know what I mean'}
      />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  // Snapshot Test
  it('renders the Card UI as expected', () => {
    const tree = renderer
      .create(
        <Card
          title='Title'
          content={'lorem ipsum something you know what I mean'}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
