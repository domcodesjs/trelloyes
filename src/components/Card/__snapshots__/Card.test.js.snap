// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders the Card UI as expected 1`] = `
<div
  className="card"
>
  <div>
    <h3>
      Title
    </h3>
    <button
      type="button"
    >
      delete
    </button>
  </div>
  <p>
    lorem ipsum something you know what I mean
  </p>
</div>
`;
